ssh-agent.c

```diff
@@ -2303,8 +2303,11 @@ main(int ac, char **av)
 
 		pidstr = getenv(SSH_AGENTPID_ENV_NAME);
 		if (pidstr == NULL) {
-			fprintf(stderr, "%s not set, cannot kill agent\n",
-			    SSH_AGENTPID_ENV_NAME);
+			execlp("systemctl", "systemctl", "--user", "stop", "ssh-agent.service", (char*)NULL);
```

Comment:
I think this introduces a footgun that currently doesn't exist: if a user logs into a system via SSH with agent forwarding enabled then they'll get a `$SSH_AUTH_SOCK` but no `$SSH_AGENT_PID`. If they then run `ssh-agent -k`, they they might kill a _different_ unrelated agent running on the host (i.e. the one started by systemd)

Reply:
That's a good point.  what do you think should be done instead?  should we just expect `ssh-agent -k` to not work for socket-activated agents?  I've never used `ssh-agent -k` in any of my regular workflows (i also don't generally forward agents, i prefer a star topology where possible), so i'm not sure what the typical expectations are.